components/scream/src/physics/p3/tests/p3_run_and_cmp.cpp

```diff
@@ -58,36 +61,43 @@ Int compare (const std::string& label, const double& tol,
 
 struct Baseline {
   Baseline () {
-    for (const bool log_predictNc : {true, false})
-      for (const int it : {1, 2})
-        params_.push_back({ic::Factory::mixed, 1800, it, log_predictNc});
+    std::string setname = "";
+    Int myit = 0;
+    for (const bool log_predictNc : {true, false}) {
+      for (const int it : {1, 2}) {
```

Comment:
@ambrad and @jgfouca ,  This is unrelated to this PR but since I went digging around in `p3_run_and_cmp` and this PR makes changes to `p3_run_and_cmp` its probably as good as any a place to ask...

Of the four cases, two just represent changing `it`, which I think just changes the `it` input in p3.  This had an impact at first inception since P3 did something different on the first step, but we have since changed that.  I compared these four cases and found no change when `it` is different.  So question 1) Should we just remove these cases and stick with the `log_predictNc` cases alone?

Follow-up, would it perhaps be beneficial to keep the two `it` cases but instead have this refer to the number of iterations of p3?  So wrap the p3_main call in a `for ps.it` loop?

Lastly, would you be opposed to changing dt from 1800 to 300?  The standard p3 timestep is 300s and I could see cases where a big dt just drives some things to zero which are then clipped and we aren't really capturing the p3 behavior in the test.

Reply:
1. Sure, remove the `it` cases if `it` no longer has effect. Should `it` be removed from `p3_main`?
2. Calling `p3_main` `it` times makes sense, too. I recommend you call `write(fid, d)` after each call so that baseline regression testing is done against multipe time slices.
3. This is synergistic with 2. You could make dt 300s and then call p3_main 6 times (say).

Reply:
Great, 
1) Yes we should probably remove it from p3_main, which I suspect can be handled in the c++ porting phase.
2) For sure, calling write(fid,d) for each step would give an even more robust baseline to test against.  I suppose we could still just have one case where we iterate 6 times as you suggested in (3).  So for clarity 2 cases total, log_predictNc true/false.  
3) see above.

I'll make the changes in this PR, unless anyone has objection.